modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -3181,34 +3179,47 @@ private void assignPartitionStates(GridDhtPartitionTopology top) {
 
             assert nodeCntrs != null;
 
-            for (int i = 0; i < nodeCntrs.size(); i++) {
-                int p = nodeCntrs.partitionAt(i);
+            if (nodeCntrs.isEmpty()) {
```

Comment:
If the `nodeCntrs` is empty we can move it right after the for-loop. I think it will look much simpler.

Reply:
Done, thanks.